stress/shell/src/main/java/alluxio/stress/cli/fuse/FuseIOBench.java

```diff
@@ -0,0 +1,365 @@
+      }
+
+      // Update thread count result
+      mThreadCountResult.setEndMs(CommonUtils.getCurrentMs());
+      mContext.mergeThreadResult(mThreadCountResult);
+
+      return null;
+    }
+
+    private void runInternal() throws Exception {
+      // When to start recording measurements
+      long recordMs = mContext.getStartMs() + FormatUtils.parseTimeSize(mParameters.mWarmup);
+      mThreadCountResult.setRecordStartMs(recordMs);
+      boolean isRead = FuseIOOperation.isRead(mParameters.mOperation);
+
+      long waitMs = mContext.getStartMs() - CommonUtils.getCurrentMs();
+      if (waitMs < 0) {
+        throw new IllegalStateException(String.format(
+            "Thread missed barrier. Set the start time to a later time. start: %d current: %d",
+            mContext.getStartMs(), CommonUtils.getCurrentMs()));
+      }
+      CommonUtils.sleepMs(waitMs);
+      mStartBarrierPassed = true;
+
+      for (int i = 0; i < mFilesPath.size(); i++) {
+        String filePath = mFilesPath.get(i);
+        while (!Thread.currentThread().isInterrupted() && (!isRead
+            || CommonUtils.getCurrentMs() < mContext.getEndMs())) {
+          long ioBytes = applyOperation(filePath);
+
+          long currentMs = CommonUtils.getCurrentMs();
+          // Start recording after the warmup
+          if (currentMs > recordMs) {
+            if (ioBytes > 0) {
+              mThreadCountResult.incrementIOBytes(ioBytes);
+              // Writing one file is done in one applyOperation call; start writing next file
+              if (!isRead) {
+                break;
+              }
+            } else if (i == mFilesPath.size() - 1) {
+              // Done reading the last file. Finish too early.
+              throw new IllegalArgumentException(String.format("Thread %d finishes reading all "
+                  + "its files before the bench ends. For more accurate result, use more files, "
+                  + "or larger files, or shorter durations."));
+            } else {
+              // Done reading this file.
+              break;
+            }
+          }
+        }
+      }
+    }
+
+    private long applyOperation(String filePath) throws IOException {
+      if (FuseIOOperation.isRead(mParameters.mOperation) && mInStream == null) {
+        mInStream = new FileInputStream(filePath);
+      }
+      switch (mParameters.mOperation) {
+        case READ: {
+          int bytesRead = mInStream.read(mBuffer);
+          if (bytesRead < 0) {
+            closeInStream();
+          }
+          return bytesRead;
+        }
+        case WRITE: {
+          mOutStream = new FileOutputStream(filePath, true);
+          long currentOffset = 0;
+          while (true) {
+            int bytesToWrite = (int) Math.min(mFileSize - currentOffset, mBuffer.length);
+            if (bytesToWrite == 0) {
+              mOutStream.close();
+              mOutStream = null;
+              break;
+            }
+            mOutStream.write(mBuffer);
```

Comment:
use `mOutStream.write(mBuffer, 0, bytesToWrite)`, otherwise if we wants to write 20 bytes, the buffer is 12 bytes, we will eventually write 24 bytes instead of 20

Reply:
Always remember to close the input/output stream when you done using it!
you can use try (FileOutStream out = .....) {} try with resources to make sure no matter what happens when writing the file, the file out stream will always be closed

Reply:
No need to use mOutStream as a member variable here